xbmc/interfaces/AnnouncementManager.cpp

```diff
@@ -248,5 +276,29 @@ void CAnnouncementManager::Announce(AnnouncementFlag flag, const char *sender, c
   if (id > 0)
     object["item"]["id"] = id;
 
-  Announce(flag, sender, message, object);
+  DoAnnounce(flag, sender, message, object);
+}
+
+void CAnnouncementManager::Process()
+{
+  SetPriority(GetMinPriority());
+
+  while (!m_bStop)
+  {
+    CSingleLock lock (m_critSection);
+    if (!m_announcementQueue.empty())
+    {
+      auto announcement = m_announcementQueue.front();
+      m_announcementQueue.pop_front();
+      {
+        CSingleExit ex(m_critSection);
+        DoAnnounce(announcement.flag, announcement.sender.c_str(), announcement.message.c_str(), announcement.item, announcement.data);
+      }
+    }
+    else
+    {
+      CSingleExit ex(m_critSection);
+      m_queueEvent.WaitMSec(1000);
```

Comment:
> Just a workaround to make it slightly less resource intensive

workaround, slightly less, are you serious? It wakes up every 1000ms for doing close to nothing, just in case. I am wondering why you didn't care in the past as it really consumed resources. Sorry, I am not in the mood for bikeshedding.